### PR TITLE
Fix VRRPv3 checksum

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- fixes VRRPv3 checksum computation https://github.com/secdev/scapy/issues/720
- Creates `in4_chksum` function: used the same way than `in6_chksum` for Upper-Layer checksum computations
- Removes duplicate chksum code (in TCP and UDP)

### CODE REVIEW

**Comment by guedou** [comment]
Thanks. Could you add a test that triggers the issue and show  that the fix works ?

**Comment by gpotter2** [comment]
That should do it

File: scapy/layers/vrrp.py [hunk 1]
```diff
@@ -10,30 +10,39 @@
...
         IntField("auth1", 0),
         IntField("auth2", 0) ]
 
     def post_build(self, p, pay):
         if self.chksum is None:
-            ck = checksum(p)
+            if self.version >= 3:
+                if isinstance(self.underlayer, IP):
+                    ck = in4_chksum(112, self.underlayer, p)
+                else:
```
Comment (guedou) [inline scapy/layers/vrrp.py]: Why not testing for IPv6 and returning a warning if the underlayer is neither IP or IPv6 ?

Comment (gpotter2) [inline scapy/layers/vrrp.py]: Right.

**Comment by gpotter2** [comment]
I'm still working on it...

**Comment by gpotter2** [comment]
Moved to https://github.com/secdev/scapy/pull/736

**Pull request closed without merging**